packages/client/src/object/Cache.test.ts

```diff
@@ -274,10 +268,10 @@ describe("AsyncCache", () => {
           i++
         ) {
           if (asyncCacheSpies.get.mock.calls[i][1] === key) {
-            expect(asyncCacheSpies.get.mock.results[i].type).toBe(
-              "throw",
+            expect(asyncCacheSpies.get.mock.settledResults[i].type).toBe(
```

Comment:
vitest in 1.x was very very strange wrt to how it handled functions that returned promises. In the old way, if the returned promise was resolved/rejected, the value itself was treated like the return value from the function. Otherwise it was treated as pending. Of course a function that returns a Promise could throw unrelated to that promise and you might want to test for that. So they changed the API to be far far more logical in terms of whats happening in the underlying system but that also broke all these carefully crafted tests.

`settledResults` is a lot like the former `results` except it only fills out AFTER the promises have been resolved/rejected. In almost all of our tests, this was a pure replacement but in some cases we had to change things.